patch/save_patch_utils.txt

```diff
@@ -0,0 +1,218 @@
+DoNothing:
+	return
+
+GetGameDataPointer:
+	set #address, GameDataStartPointers
+.get_pointer_loop
+	gethalfwordinc #temp, #address
+	ifge #temp, #curbuild, .got_pointer
+	add #address, 8
+	jump .get_pointer_loop
+
+.got_pointer
+	getwordinc #temp, #address
+	add #result, #1, #temp
+	ifeq #savefile, 1, .done
+	getword #temp, #address
+	add #result, #1, #temp
+.done
+	return
+
+GetGameDataOffsetConstant:
+; (0 = items, 1 = medicine, 2 = balls, 3 = berries, 4 = PC items, 5 = party count, 6 = daycare man, 7 = daycare lady, error if >7)
+; returns targeted constant from OldPocketOffsets
+	set #address, PocketOffsetsByBuild
+.get_offset_loop
+	gethalfwordinc #temp, #address
+	jumpz #temp, .error_build
+	ifge #temp, #curbuild, .get_constant_loop
+	add #address, 18
+	jump .get_offset_loop
+
+.get_constant_loop
+	gethalfwordinc #result, #address
+	jumpz #result, .error_oob
+	retz #1
+	decrement #1
+	jump .get_constant_loop
+
+.error_build
+	bufstring .error_build_string_1
+	bufnumber #curbuild
+	bufstring .error_build_string_2
+	printbuf
+	exit 1
+
+.error_oob
+	print .error_oob_string
+	exit 1
+
+.error_build_string_1
+	string "ERROR: Invalid build " 
+.error_build_string_2
+	string " for GetGameDataOffsetConstant"
+
+.error_oob_string
+	string "ERROR: GetGameDataOffsetConstant went out-of-bounds"
+
+ShiftFlags:
+	; thank you kindly to ax6 for creating the base of this function
+	; in: #1: flag array size (in bits), #2: pointer to mapping table
+	; file pointer must be at the beginning of the array
+	; the mapping table is a halfword array of upper bound index (exclusive), offset; end with -1
+	pushpos
+	add #1, 31
+	shiftright #1, 5
+	shiftleft #3, #1, 2
+	add #3, 4
+	seekfwd #3
+	set #3, #1
+.readloop
+	seekback 8
+	readword #4
+	push #4
+	decrement #3
+	jumpnz #3, .readloop
+	stackread #address, #1
+	seek #address
+
+.fixloop
+	; #3 = current index, #4 = offset, #5 = upper bound index, #6 = stack pos
+	gethalfwordinc #5, #2
+	ifeq #5, 0xffff, .done
+	gethalfwordinc #4, #2
+	jumpz #4, .next
+.innerloop
+	shiftright #source, #3, 5
+	multiply #source, 4
+	add #source, #address
+	seek #source
+	getfileword #source ; word to read from
+	and #temp, #3, 31
+	shiftright #source, #temp
+	or #source, -2 ; get source bit, preserve other target bits
+	add #6, #3, #4
+	and #6, 0xffff
+	and #temp, #6, 31
+	subtract #temp, 32, #temp
```

Comment:
`subtract #temp, #zero, #temp` would be shorter and achieve the same effect. Remember that shift counts only care about the lower five bits.